### PR TITLE
HTML table include histograms.

### LINKED ISSUE #18: Add histograms and diff information to HTML table output
Currently the HTML tables show numerical information only (see the screenshot here: https://github.com/softdevteam/warmup_experiment/pull/332 ). Ideally these should show the same information as the LaTeX tables, i.e. histograms and (for diff tables) delta and improved / worsened information.

For the histograms we probably want a JS library just as d3, which has a histogram ( [docs](https://github.com/d3/d3-array/blob/master/README.md#histograms) [example](https://bl.ocks.org/mbostock/3048450) ).

### PR DESCRIPTION
**Pull request by snim2**
Fixes #18 

Needs squashing.

### Example results table:

PDF: [warmup_table.pdf](https://github.com/softdevteam/warmup_stats/files/1713691/warmup_table.pdf)
HTML: screenshot from 2018-02-10 19-21-57

### Example diff table:

PDF: [tom2.pdf](https://github.com/softdevteam/warmup_stats/files/1713696/tom2.pdf)
HTML: screenshot from 2018-02-10 19-20-55

### CODE REVIEW

**Comment by ltratt** [comment]
This looks pretty good already. One comment: should/could we make the rows the same height? At the moment rows without histograms are much shorter. I think the rows with histograms have got a fair bit of spare vertical space, so perhaps we can make the short ones a bit taller and the tall ones a bit shorter?

**Comment by snim2** [comment]
OK, I have lost the will to live trying to fix the vertical alignment in the numerical bits of the tables, but otherwise this is looking a little neater:

screenshot from 2018-02-10 21-36-11
screenshot from 2018-02-10 21-36-14

File: warmup/summary_statistics.py [hunk 1]
```diff
@@ -465,9 +465,9 @@ def colour_html_cell(result, text, align=None):
         text_align = ''
     return '<td %s %s>%s</td>' % (text_align, colour, text)
 
-
+# <span class="vertical_aligner"></span>
```
Comment (ltratt) [inline warmup/summary_statistics.py]: Commented line?

**Comment by ltratt** [comment]
You could, perhaps put a div/table inside the cell to force vertical alignment?

**Comment by snim2** [comment]
Yeah, tried that (it's the `tdtext` bit of the CSS) but for some reason no combination of magic incantations has worked.

**Comment by ltratt** [comment]
OK, let's merge this PR as-is, because it's pretty much good enough, but maybe ask @ptersilie for advice (he knows HTML better than I do!) when he's next around?

I guess some squashing is needed first though!

**Comment by snim2** [comment]
OK, squashed!

**Pull request merged**